doc/langref.html.in

```diff
@@ -4034,7 +4034,7 @@ test "access variable after block scope" {
     x += 1;
 }
       {#code_end#}
-      <p>Blocks are expressions. When labeled, {#syntax#}break{#endsyntax#} can be used
+      <p>Blocks are statements, not expressions. When labeled, {#syntax#}break{#endsyntax#} can be used
```

Comment:
Labeled blocks *are* expressions, so
```suggestion
      <p>When labeled, blocks are expressionss and {#syntax#}break{#endsyntax#} can be used
```

Reply:
It is not true that unlabelled blocks are always statements as they can be used of the right side of an assignment and result in a type of `void`

It is also not true that labelled blocks are always expressions, the label can be used only for `break` and `continue` control flow without any return value.

```zig
pub fn main() void {
    const a = {
        // this block is unlabelled but it is an expression and it evaluates to `void`
    };
    @compileLog(@TypeOf(a));

    blk: {
        // this block is labelled but is not an expression
        break :blk;
    }
}
```

Reply:
Hi, I'm very new to Zig so this may be a silly question, but is the block evaluating to void construct useful?  Writing something like that seems like a mistake on the programmers side (and the compiler shouldn't allow it).

Also please take my comments with a grain of salt, I'm very new at Zig. Actually @zenspider was helping me try to understand why my `if` expression wasn't working, and it turned out to be due to the fact that blocks aren't expressions (IOW `const x = if () { ... }` is very different than `const x = if () ...`, and I didn't understand that)

Reply:
> Hi, I'm very new to Zig so this may be a silly question, but is the block evaluating to void construct useful? Writing something like that seems like a mistake on the programmers side (and the compiler shouldn't allow it).

Its the only way to initialize a void field. This is common practice in particular if you're using comptime to turn a field "on" or "off". i.e.:

```zig
const Widget = struct {
  const KnobType = if (builtin.target.isDarwin()) u64 else void;
  const KnobDefault = if (builtin.target.isDarwin()) 0 else {};

  knobs: KnobType = KnobDefault,
}
```

A void field takes up no space.

Reply:
Actually, it is one of two ways to initialize a void value. The second way is the void literal '`void{}`'.

Reply:
> > Hi, I'm very new to Zig so this may be a silly question, but is the block evaluating to void construct useful? Writing something like that seems like a mistake on the programmers side (and the compiler shouldn't allow it).
> 
> Its the only way to initialize a void field. This is common practice in particular if you're using comptime to turn a field "on" or "off". i.e.:
> 
> ```zig
> const Widget = struct {
>   const KnobType = if (builtin.target.isDarwin()) u64 else void;
>   const knob_default = if (builtin.target.isDarwin()) 0 else {};
> 
>   knobs: KnobType = knob_default,
> }
> ```
> 
> A void field takes up no space.

Awesome, this makes a lot of sense.  Thank you for taking the time to explain, I really appreciate it!!